lib/fpm/package.rb

```diff
@@ -331,17 +331,35 @@ def template(path)
     return erb
   end # def template
 
-  def to_s(fmt="NAME.TYPE")
-    fmt = "NAME.TYPE" if fmt.nil?
-    fullversion = version.to_s
-    fullversion += "-#{iteration}" if iteration
-    return fmt.gsub("ARCH", architecture.to_s) \
-      .gsub("NAME", name.to_s) \
-      .gsub("FULLVERSION", fullversion) \
-      .gsub("VERSION", version.to_s) \
-      .gsub("ITERATION", iteration.to_s) \
-      .gsub("EPOCH", epoch.to_s) \
-      .gsub("TYPE", type.to_s)
+  #######################################
+  # The following methods are provided to
+  # easily override particular substitut-
+  # ions performed by to_s for subclasses
+  #######################################
+  def to_s_arch; architecture.to_s; end
+  def to_s_name; name.to_s;         end
+  def to_s_fullversion
+    ver = to_s_version
+    ver += "-#{iteration}" if iteration
```

Comment:
Maybe. Or, should we leave it as is and use ``type.to_s`` in the ``to_s_extension`` below? Maybe this second approach would be better, as it does not introduce "unexpected" interdependences between ``to_s_*`` functions?